modules/4337/contracts/Safe4337Module.sol

```diff
@@ -222,10 +261,20 @@ contract Safe4337Module is IAccount, HandlerContext, CompatibilityFallbackHandle
      */
     function _validateSignatures(PackedUserOperation calldata userOp) internal view returns (uint256 validationData) {
         (bytes memory operationData, uint48 validAfter, uint48 validUntil, bytes calldata signatures) = _getSafeOp(userOp);
-        try ISafe(payable(userOp.sender)).checkSignatures(keccak256(operationData), operationData, signatures) {
+
+        // The `checkSignatures` function in the Safe contract does not force a fixed size on signature length.
+        // A malicious bundler can pad the Safe operation `signatures` with additional bytes, causing the account to pay more gas than needed for user operation validation (capped by `verificationGasLimit`).
+        // `_checkSignatureLength` ensures that there are no additional bytes in the `signature` than are required.
+        bool validSignature = _checkSignatureLength(signatures, ISafe(payable(userOp.sender)).getThreshold());
+
+        try ISafe(payable(userOp.sender)).checkSignatures(keccak256(operationData), operationData, signatures) {} catch {
+            validSignature = false;
+        }
+
+        if (validSignature) {
             // The timestamps are validated by the entry point, therefore we will not check them again
             validationData = _packValidationData(false, validUntil, validAfter);
-        } catch {
+        } else {
             validationData = _packValidationData(true, validUntil, validAfter);
         }
```

Comment:
I just realized, but this can be simplified to:

```solidity
        validationData = _packValidationData(!validSignature, validUntil, validAfter);
```

Reply:
https://github.com/safe-global/safe-modules/pull/453/commits/2743e53077f87e34811069ff1f2595d6a4f75492